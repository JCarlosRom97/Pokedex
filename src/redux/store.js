import { configureStore } from "@reduxjs/toolkit";
import PokemonReducer from "./reducers/index.reducer";
import detailReducer from "./reducers/detail.reducer";
export default configureStore({
    reducer: {
        PokemonDataList: PokemonReducer,
        detail: detailReducer
    },
})

