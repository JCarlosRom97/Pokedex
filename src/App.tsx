import React, { useEffect, useState } from "react";
import { useSelector, useDispatch } from "react-redux";
import { add } from "./redux/reducers";
import Axios  from "axios";
import PokemonImage from './Images/pokeapi_256.png'
import sampleImage from './Images/bulbasur.png'
import Table from "./Components/Table";
import { Column, Container, ContainerImages, Image } from "./Styles";
import { state } from "./types";

const App = () => {
  const state = useSelector((state: state) => state.example.data );
  const dispatch = useDispatch();  
  const [imagePokemon, setImagePokemon] = useState(sampleImage);

  useEffect(()=>{
    async function getData() {
      const pokemonData = await Axios.get('https://pokeapi.co/api/v2/pokemon/')
      dispatch(add(pokemonData.data.results))
    }
    getData()
  },[])

  return (
    <Container>
      <Column columns={5}>
        <ContainerImages>
          <Image src={PokemonImage}/>
          <Image src={imagePokemon} style={{width:250}}/>
        </ContainerImages>
      </Column>
      <Column columns={7}>
        <Table pokemonData={state} onChangeImage={(imgUrl)=> setImagePokemon(imgUrl)}/>
      </Column>
    </Container>
  );
}

export default App;
