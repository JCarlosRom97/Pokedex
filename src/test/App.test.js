import renderer from "react-test-renderer"
import App from "../App"
import { Provider } from "react-redux"
import store from "../redux/store"
import { BrowserRouter } from "react-router-dom"
import { render, screen, waitFor } from '@testing-library/react'
import * as axios from "axios";

jest.mock('axios');

const mockUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
   ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockUsedNavigate,
}));


describe('App', () => {
    
    beforeEach(() => {
        jest.resetAllMocks()
    })

    
    test('snapshot should be the same', () =>{
        const component = renderer.create(
            <Provider store={store}>
                <BrowserRouter>
                    <App/>
                </BrowserRouter>
            </Provider>
        )
    
        let tree = component.toJSON();
    
        expect(tree).toMatchSnapshot()
    })

    test('renders users when API call succeeds', async () => {
        
        const fakePokemons = [
            { name: 'bulbasaur', url:'https://pokeapi.co/api/v2/pokemon/1/' },
            { name: 'ivysaur',  url:'https://pokeapi.co/api/v2/pokemon/2/'},
        ]

        axios.get.mockResolvedValue({
            data: {
                results: fakePokemons
            }
        });

        const {container, asFragment} = render(
            <Provider store={store}>
                <BrowserRouter>
                    <App/>
                </BrowserRouter>
            </Provider>
        )

        const listNode = await waitFor(() => container.querySelector('tbody'));
        expect(listNode.children).toHaveLength(2);
        expect(asFragment()).toMatchSnapshot();
    
    })

    test('renders error when API call fails', async () => {})

})