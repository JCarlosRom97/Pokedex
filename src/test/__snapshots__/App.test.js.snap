// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders users when API call succeeds 1`] = `
<DocumentFragment>
  <div
    class="sc-gtsqUy dUhiaT"
  >
    <div
      class="sc-hKFymg iQaxrS"
      columns="5"
    >
      <div
        class="sc-dlniIP dKGwVR"
      >
        <img
          class="sc-eCAqax bPzBCX"
          src="pokeapi_256.png"
        />
        <img
          class="sc-eCAqax bPzBCX"
          src="bulbasur.png"
          style="width: 250px;"
        />
      </div>
    </div>
    <div
      class="sc-hKFymg iQaxqI"
      columns="7"
    >
      <table
        class="sc-bdnyFh CUoeZ"
      >
        <tbody>
          <tr>
            <td>
              bulbasaur
            </td>
            <td>
              <img
                class="sc-eCAqax dGuNbV"
                height="20"
                src="pokeball.png"
              />
            </td>
          </tr>
          <tr>
            <td>
              ivysaur
            </td>
            <td>
              <img
                class="sc-eCAqax dGuNbV"
                height="20"
                src="pokeball.png"
              />
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</DocumentFragment>
`;

exports[`App snapshot should be the same 1`] = `
<div
  className="sc-gtsqUy dUhiaT"
>
  <div
    className="sc-hKFymg iQaxrS"
    columns={5}
  >
    <div
      className="sc-dlniIP dKGwVR"
    >
      <img
        className="sc-eCAqax bPzBCX"
        src="pokeapi_256.png"
      />
      <img
        className="sc-eCAqax bPzBCX"
        src="bulbasur.png"
        style={
          Object {
            "width": 250,
          }
        }
      />
    </div>
  </div>
  <div
    className="sc-hKFymg iQaxqI"
    columns={7}
  >
    <table
      className="sc-bdnyFh CUoeZ"
    >
      <tbody />
    </table>
  </div>
</div>
`;

exports[`snapshot should be the same 1`] = `
<div
  className="sc-gtsqUy dUhiaT"
>
  <div
    className="sc-hKFymg iQaxrS"
    columns={5}
  >
    <div
      className="sc-dlniIP dKGwVR"
    >
      <img
        className="sc-eCAqax bPzBCX"
        src="pokeapi_256.png"
      />
      <img
        className="sc-eCAqax bPzBCX"
        src="bulbasur.png"
        style={
          Object {
            "width": 250,
          }
        }
      />
    </div>
  </div>
  <div
    className="sc-hKFymg iQaxqI"
    columns={7}
  >
    <table
      className="sc-bdnyFh CUoeZ"
    >
      <tbody />
    </table>
  </div>
</div>
`;
